<?xml version="1.0"?>
<robot name="motor_robot">

  <link name="base_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="0.001"/>
      <inertia ixx="1.6667e-08" ixy="0.0" ixz="0.0"
               iyy="1.6667e-08" iyz="0.0"
               izz="1.6667e-08"/>
    </inertial>
  </link>

  <joint name="motor_joint" type="continuous">
    <parent link="base_link"/>
    <child link="motor_link"/>
    <origin xyz="0 0 0.1" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <link name="motor_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="0.2"/>
      <inertia ixx="0.0007916667" ixy="0.0" ixz="0.0"
               iyy="0.0007916667" iyz="0.0"
               izz="0.00025"/>
    </inertial>

    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder radius="0.05" length="0.2"/>
      </geometry>
    </collision>

    <visual>
      <geometry>
        <cylinder radius="0.05" length="0.2"/>
      </geometry>
      <material name="gray">
        <color rgba="0.6 0.6 0.6 1.0"/>
      </material>
    </visual>
  </link>

  <!-- ros2_control -->
  <ros2_control name="GazeboSystem" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>
    <joint name="motor_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>

  <!-- Plugin Gazebo Classic -->
  <gazebo>
    <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
      <parameters>/home/luis/tesis_ws/src/quanser_digital_twin/config/controllers.yaml</parameters>
    </plugin>
  </gazebo>

</robot>

